utils/utils.py

```diff
@@ -213,21 +219,29 @@ def map_to_hemisphere(s: np.ndarray, n: np.ndarray, e=1.):
 
     vec = u * sin_theta * cos_psi + v * sin_theta * sin_psi + n * cos_theta
     norm_vec = vec / np.linalg.norm(vec)
+    pdf = norm_vec[1] / np.pi
+    light = cartesian_to_spherical(norm_vec)
     # Return the result
-    return norm_vec
+    return light, pdf
 
+def uniform(eps: np.ndarray):
+    eps[0] = math.acos(eps[0])
+    eps[1] *= 2 * math.pi
+    pdf = 1 / (2 * np.pi)
+    return eps, pdf
 
 # Getting samples similar to HIBRID
 def get_test_samples(points: np.ndarray):
 
     norm_pts = points[:, 3:]
-    lights = np.zeros(norm_pts.shape, dtype=np.float32)
+    lights = np.zeros([norm_pts.shape[0], 2], dtype=np.float32)
     pdfs = np.zeros(norm_pts.shape[0], dtype=np.float32)
 
     for i in range(norm_pts.shape[0]):
         s = np.random.uniform(0., 1., 2)
-        norm = norm_pts[i]
-        light = map_to_hemisphere(s, norm)
+        norm = np.array([0, 1, 0])
```

Comment:
(0, 1, 0) - which axis do we use here? y-axis?

Reply:
This should be clarified with Sergei.